gst/nnstreamer/ml_agent.c

```diff
@@ -12,34 +12,42 @@
  * @bug     No known bugs except for NYI items
  */
 
+#include <json-glib/json-glib.h>
+#include <nnstreamer_log.h>
 #include <ml-agent-interface.h>
 
 #include "ml_agent.h"
 
 const gchar URI_SCHEME[] = "mlagent";
 const gchar URI_KEYWORD_MODEL[] = "model";
+const gchar JSON_KEY_MODEL_PATH[] = "path";
 
 /**
  * @brief Parse the given URI into the valid file path string
+ * @param[in] val A pointer to a GValue holding a G_TYPE_STRING value
+ * @return A newly allocated c-string containing the model file path in the case that the valid URI is given.
+ * Otherwise, it simply returns c-string that the val contains.
+ * @note The caller should free the return c-string after using it.
```

Comment:
Specify how errors will be handled? (NULL return? ERRPTR? some special strings? ...)